### PR TITLE
Fixed issue where sample name and checkbox fields were editable

### PR DESCRIPTION
**Pull request by joshsadam**
Signed-off-by: Josh Adam <josh.adam@canada.ca>

### CODE REVIEW

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/linelist/LineListController.java [hunk 1]
```diff
@@ -349,4 +359,18 @@ This field is to display to the user any notification icons that they might have
...
+	 *
+	 * @param project {@link Project}
+	 * @return {@link Boolean} true if user can edit on the current project
+	 */
+	private boolean canUserEdit(Project project) {
+		Authentication authentication = SecurityContextHolder.getContext()
+				.getAuthentication();
+		User user = (User) authentication.getPrincipal();
+		return user.getSystemRole()
+				.equals(Role.ROLE_ADMIN) || projectOwnerPermission.isAllowed(authentication, project);
```
Comment (tom114) [inline src/main/java/ca/corefacility/bioinformatics/irida/ria/web/linelist/LineListController.java]: Should be no need to add the `ROLE_ADMIN` check here.  The permission will return true if the user is an admin (and the project is local).

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/linelist/LineListController.java [hunk 2]
```diff
@@ -211,7 +218,8 @@ private AgGridColumn mapFieldToColumn(MetadataTemplateField field) {
 			int index = allFieldsLabels.indexOf(field.getFieldKey());
 			allFieldsAgGridColumns.remove(index);
 			allFieldsLabels.remove(index);
-			templateAgGridColumns.add(mapFieldToColumn(field));
+			// NEed to add parameter for if they have permissions to edit.
```
Comment (tom114) [inline src/main/java/ca/corefacility/bioinformatics/irida/ria/web/linelist/LineListController.java]: `NEed` typo

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/linelist/LineListController.java (hunk 1 above)
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/linelist/LineListController.java]: Removed in b5e9275e013b023cb27469613c83b11cbc747539

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/linelist/LineListController.java (hunk 2 above)
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/linelist/LineListController.java]: Fixed in 090463c35e2e98f0fdd78c6b1c7b7e3bda3bb59a

**Comment by joshsadam** [comment]
FIxed sample edit icon in c153bd5d85e0e437b3dbc09f3ac09eff24a6a2e3

**Pull request merged**